### PR TITLE
Added e2e tests for list

### PR DESCRIPTION
**Pull request by gapra-msft**

### CODE REVIEW

**Review by siminsavani-msft — APPROVED** [review]
LGTM!

File: e2etest/zt_newe2e_list_test.go [hunk 1]
```diff
@@ -45,9 +53,261 @@ func (s *ListSuite) Scenario_ListBasic(svm *ScenarioVariationManager) {
...
+				srcContainer.(RemoteResourceManager).WithSpecificAuthType(EExplicitCredentialType.SASToken(), svm, CreateAzCopyTargetOptions{
+					SASTokenOptions: GenericServiceSignatureValues{
+						ContainerName: srcContainer.ContainerName(),
+						Permissions:   (&blobsas.ContainerPermissions{Read: true, List: true}).String(),
+					},
+				}),
+			},
+			Flags: ListFlags{
+				GlobalFlags: GlobalFlags{
+					OutputType: to.Ptr(common.EOutputFormat.Json()),
```
Comment (adreed-msft) [inline e2etest/zt_newe2e_list_test.go:115]: JSON should be default these days, I think? Check the flags file.

**Review by adreed-msft — APPROVED** [review]

File: e2etest/zt_newe2e_list_test.go (hunk 1 above)
Comment (gapra-msft) [reply e2etest/zt_newe2e_list_test.go:115]: yeah. I think there used to be a bug with the default output type. seems to be working fine now. I'll remove these.

File: e2etest/zt_newe2e_list_test.go [hunk 2]
```diff
@@ -45,9 +53,261 @@ func (s *ListSuite) Scenario_ListBasic(svm *ScenarioVariationManager) {
...
+			Body:       NewRandomObjectContentContainer(svm, SizeFromString("1K")),
+		})
+		props := obj.GetProperties(svm)
+		versionId := common.IffNotNil(props.BlobProperties.VersionId, "")
+		expectedObjects[AzCopyOutputKey{Path: blobName, VersionId: versionId}] = cmd.AzCopyListObject{
+			Path:             blobName,
+			ContentLength:    "1.00 KiB",
+			LastModifiedTime: props.BlobProperties.LastModifiedTime,
+			VersionId:        versionId,
+			BlobType:         common.IffNotNil(props.BlobProperties.Type, ""),
```
[the remark this answers is not in the document]
Comment (gapra-msft) [reply e2etest/zt_newe2e_list_test.go:145]: don't they still get output in the list output?

**Pull request merged**